File: benchmark/c/cpw/count_distinct_non_negative_pairs_x_y_satisfy_inequality_xx_yy_n_2/count_distinct_non_negative_pairs_x_y_satisfy_inequality_xx_yy_n_2.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int count_distinct_non_negative_pairs_x_y_satisfy_inequality_xx_yy_n_2 ( int n ) ;
int count_distinct_non_negative_pairs_x_y_satisfy_inequality_xx_yy_n_2 ( int n ) {
  int res = 0;
  for ( int x = 0;
  x * x < n;
  x ++ ) for ( int y = 0;
  x * x + y * y < n;
  y ++ ) res ++;
  return res;
}

File: translation_result/else_bms/success/c/cpw/count_distinct_non_negative_pairs_x_y_satisfy_inequality_xx_yy_n_2/gemini/BaseRepair/count_distinct_non_negative_pairs_x_y_satisfy_inequality_xx_yy_n_2.rs

fn count_distinct_non_negative_pairs_x_y_satisfy_inequality_xx_yy_n_2(n: i32) -> i32 {
    let mut res = 0;
    for x in 0.. {
        if x * x >= n {
            break;
        }
        for y in 0.. {
            if x * x + y * y >= n {
                break;
            }
            res += 1;
        }
    }
    res
}
